### PR TITLE
Add information linear and non-linear solver outputs

### PR DESCRIPTION
**Pull request by hepap**
# Description of the problem

- The console output of the non-linear solver in the NS solver was showing the the norms of the newton update without distinction between the velocity and pressure contributions.
- The final residual norm of the linear solver was not outputted, only the tolerance.

# Description of the solution

- Add the distinction between the velocity and pressure contributions in the non-linear solver output.
- Add the final value of the linear residual when verbosity is at verbose. If it is at extra verbose, the solver output the residual for each iteration (already implemented but not documented).

# How Has This Been Tested?

A unit test was added.

- [x] vector_problem.cc

# Documentation

Add a comment for the extra verbose verbosity of the linear solver.

- [x] linear_solver_control.rst

### CODE REVIEW

File: source/solvers/navier_stokes_base.cc
Comment (blaisb) [inline source/solvers/navier_stokes_base.cc:2522]: maybe it's better to use std::max here no?
like local_max = std::max(local_max,dof_newton_update);

File: source/solvers/navier_stokes_base.cc
Comment (blaisb) [inline source/solvers/navier_stokes_base.cc:2545]: same as above :)

File: source/solvers/navier_stokes_base.cc
Comment (blaisb) [inline source/solvers/navier_stokes_base.cc:2578]: It's funny how much easier it is with the blocks heh ?%

File: tests/core/vector_problem.cc
Comment (blaisb) [inline tests/core/vector_problem.cc:111]: std::max here also

File: tests/core/vector_problem.cc
Comment (blaisb) [inline tests/core/vector_problem.cc:98]: should it not be instead -DBL_MAX ?

File: source/solvers/navier_stokes_base.cc
Comment (blaisb) [inline source/solvers/navier_stokes_base.cc:2507]:
```suggestion
      double local_max = -std::numeric_limits<double>::max();
```
is a better way to write this. The max you want it to start at -max, this way 0 can still be a max :)

File: tests/core/non_linear_test_system_01.h
Comment (blaisb) [inline tests/core/non_linear_test_system_01.h:136]: Should the test output this also? Could be interesting if it is portable.

**Review by blaisb — CHANGES_REQUESTED** [review]
Very few comments, amazing work @hepap

File: source/solvers/navier_stokes_base.cc
Comment (PierreLaurentinCS) [reply source/solvers/navier_stokes_base.cc:2507]: One can even use std::numeric_limits<double>::lowest() to avoid the -max notation

File: source/solvers/navier_stokes_base.cc
Comment (PierreLaurentinCS) [inline source/solvers/navier_stokes_base.cc:2532]:
```suggestion
      local_max = -std::numeric_limits<double>::max();
```
just in case

**Review by PierreLaurentinCS — COMMENTED** [review]
Amazing feature, and very few comments. Good job !

File: source/solvers/navier_stokes_base.cc
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_base.cc:2523]: These loops on index_sets could be inlined if you wish :)

File: source/solvers/navier_stokes_base.cc
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_base.cc:2565]:
```suggestion
  else if constexpr (std::is_same_v<VectorType, GlobalBlockVectorType>)
```

**Review by AmishgaAlphonius — APPROVED** [review]
Great work ! Thanks ! :)
I only have a few minor comments.

File: source/solvers/navier_stokes_base.cc
Comment (blaisb) [reply source/solvers/navier_stokes_base.cc:2565]: I don't see the point of the else here.  it's actually better to have an undefined behavior than have an else taht could maybe not be an else (other than if you throw at the end something in an else)

**Comment by hepap** [comment]
I think it is ready to merge if the tests pass! The output was add for CH as well :)

**Review by blaisb — APPROVED** [review]
Once CI is fixed, this is ready for a merge IMO

**Comment by hepap** [comment]
Should be alright now!

**Pull request merged**
